shared/src/utils/validators/is-valid-object.ts

```diff
@@ -0,0 +1,10 @@
+import { plainToInstance } from 'class-transformer';
+import { validateSync } from 'class-validator';
+import type { Constructor } from '../constructor';
+
+export function isValidObject<T extends object>(
+    type: Constructor<T>,
+    value: unknown
+): boolean {
```

Comment:
```suggestion
): value is T {
```

I don't think this will be of much use, but at least it clarifies the intention of this function even more.